types/convert.go

```diff
@@ -242,9 +242,30 @@ func getValidIntPrefix(sc *stmtctx.StatementContext, str string) (string, error)
 // floatStrToIntStr converts a valid float string into valid integer string which can be parsed by
 // strconv.ParseInt, we can't parse float first then convert it to string because precision will
 // be lost.
```

Comment:
> we can't parse float first then convert it to string because precision will be lost.

The float to int conversion should indeed lose some precision, why we cannot call `strconv.ParseFloat` directly?

Reply:
If directly call `strconv.ParseFloat`, the string value "18446744073709551615" which is the max number of unsigned int will cause some precision to lose.

Reply:
Maybe we can put this example into comment